### PR TITLE
Managed multiresultvec

### PR DESCRIPTION
**Pull request by andrei-marinica**
Implemented:
ManagedVarArgs
ManagedMultiResultVec
(they are the same type)

### CODE REVIEW

File: elrond-wasm/src/types/managed/managed_multi_result_vec.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    T: ManagedVecItem<M> + TopDecode,
+{
+    pub fn to_vec(&self) -> ManagedVec<M, T> {
+        let mut result = ManagedVec::new_empty(self.raw_buffers.type_manager());
+        let serializer = ManagedSerializer::new(self.raw_buffers.type_manager());
+        for item in self.raw_buffers.into_iter() {
+            result.push(serializer.top_decode_from_managed_buffer(&item));
+        }
+        result
+    }
```
Comment (dorin-iancu) [inline elrond-wasm/src/types/managed/managed_multi_result_vec.rs:92]: For consistency, methods with the naming `to_other_type` should consume `self`:
```suggestion
    pub fn to_vec(self) -> ManagedVec<M, T> {
        let mut result = ManagedVec::new_empty(self.raw_buffers.type_manager());
        let serializer = ManagedSerializer::new(self.raw_buffers.type_manager());
        for item in self.raw_buffers.into_iter() {
            result.push(serializer.top_decode_from_managed_buffer(&item));
        }
        result
    }
```

Maybe also add some rust-docs explaining that this is an expensive operation. It's not immediately obvious, as in the previous version it was a NOP.

File: mandos/src/esdt.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+use super::*;
+use std::collections::BTreeMap;
+
+#[derive(Debug)]
+pub struct Esdt {
+    pub token_identifier: Option<BytesValue>,
```
Comment (dorin-iancu) [inline mandos/src/esdt.rs:6]: Should this be an Option?

**Review by dorin-iancu — APPROVED** [review]

File: elrond-wasm/src/types/managed/managed_multi_result_vec.rs (hunk 1 above)
Comment (andrei-marinica) [reply elrond-wasm/src/types/managed/managed_multi_result_vec.rs:92]: "into_..." consumes type,, "to_..." does not

File: mandos/src/esdt.rs (hunk 2 above)
Comment (andrei-marinica) [reply mandos/src/esdt.rs:6]: I think Alin committed this here by accident. It's ok, we'll return on this.

**Pull request merged**